### ISSUE TITLE
Obfuscation and usage

### ISSUE THREAD

**Issue by kornrunner** [thoughtpolice/libphutil-yubikey#1]
Hi,

I've tried your extension, it works great, thanks! 

Just wanted to mention a thing or two that could be changed in the implementation:

- OTP input should be obfuscated (password input vs current plain text input)
- I believe only one 2FA input should be sufficient. On my install, I've already got TOTP, and after I've added Yubikey - I have to enter both codes to enter high security mode

What do you think?

**Comment by thoughtpolice** [comment]
That first one can be fixed I think by tweaking the call to `AphrontFormTextControl`, yes. Thanks for the suggestion!

I don't think the second one is so easy. That's because Phabricator [explicitly runs every configured MFA token](https://github.com/phacility/phabricator/blob/2befd239a85ea696e1cec8abc5f2eb24927e91af/src/applications/auth/engine/PhabricatorAuthSessionEngine.php#L420) on login, by design (I mean, that *could* be a mistake, but that code is pretty clear and doesn't seem like it).

I honestly imagine this in particular is intentional in the design, though it may not be written down. The current Phabricator MFA support is very, very simple. Ideally, Phabricator would really have some kind of policy flow between MFA tokens I think, that would allow custom recovery or progress flows; e.g. "Allow login with Yubikey first. If successful, continue. If unsuccessful (I lost my keys due to a bear attack), allow TOTP, or SMS tokens." That would let you set up all kinds of configurations and relations between tokens.

However, this is probably Not Going To Happen for a Really Really Long Time And In Fact, Probably Never Happen. The main reason being, it's complex to implement the above mechanism (e.g. how does it interact with "Require MFA for All Users"? How do you merge user policies with global policies, etc). But also, because your administrator can always trivially unlock your account, so not having a backup isn't *incredibly* damning, at the moment. In most organizations this probably isn't a *gigantic* hurdle while you wait for your sysadmin to get coffee.

So, in all reality, I imagine this *probably* won't change at all. And I imagine upstream would likely reject a bug report (unless it's carefully considered and written). At least not until upstream also starts accepting other MFA schemes; at the moment they still haven't accepted this one (despite the fact I think it'd actually be pretty useful!) nor anything like U2F, so the demand really isn't there to fix it, either. There are maybe like 2 or 3 installs using this extension, so it's just not a priority. Unfortunately there's really nothing I can do about this.
